### PR TITLE
dvc: rename "working tree" to "workspace"

### LINKED ISSUE #3892: brancher: rename `working tree` -> `workspace`
Much more natural term. Was also noted by @dmpetrov while working with plots. Needed for cml release.

### PR DESCRIPTION
**Pull request by efiop**
Fixes #3892

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. If the CLI API is changed, I have updated [tab completion scripts](https://github.com/iterative/dvc/tree/master/scripts/completion).

There are only two occurrences of `working tree` in docs:
```
./content/docs/tutorials/pipelines.md:working tree:                                                                     
./content/docs/tutorials/deep/sharing-data.md:references in the <abbr>workspace</abbr> (in the _working tree_ of the Git
```
neither of those need to be changed right now.

* [x] ❌ I will check DeepSource, CodeClimate, and other sanity checks below. (We consider them recommendatory and don't expect everything to be addressed. Please fix things that actually improve code or fix bugs.)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/repo/brancher.py [hunk 1]
```diff
@@ -30,10 +30,10 @@ def brancher(  # noqa: E302
     scm = self.scm
 
     self.tree = WorkingTree(self.root_dir)
-    yield "working tree"
+    yield "workspace"
 
-    if revs and "working tree" in revs:
-        revs.remove("working tree")
+    if revs and "workspace" in revs:
```
Comment (Suor) [inline dvc/repo/brancher.py:35]: There actually can be a tag or branch named `workspace` so this is no longer safe.

Comment (Suor) [reply dvc/repo/brancher.py:35]: This will also clash in many other places like plots and metrics.

Comment (skshetry) [reply dvc/repo/brancher.py:35]: @Suor, is there any need for `workspace in revs` check at all?

Comment (Suor) [reply dvc/repo/brancher.py:35]: Making it a flash won't fix output.

**Comment by jorgeorpinel** [comment]
Docs updated in https://github.com/iterative/dvc.org/pull/1382/commits/c73f7c0aee764d2b906da901c4b0c5a78a31ba87

**Pull request merged**